logstash-core/lib/logstash/instrument/periodic_poller/jvm.rb

```diff
@@ -1,13 +1,92 @@
 # encoding: utf-8
 require "logstash/instrument/periodic_poller/base"
+require 'monitoring'
 
 module LogStash module Instrument module PeriodicPoller
   class JVM < Base
+
+    attr_reader :heap_metrics, :non_heap_metrics, :pools_metrics
+
     def initialize(metric, options = {})
       super(metric, options)
+      jvm_metrics      = metric.namespace(:jvm)
+      memory_metrics   = jvm_metrics.namespace(:memory)
+      @heap_metrics     = memory_metrics.namespace(:heap)
+      @non_heap_metrics = memory_metrics.namespace(:non_heap)
+      @pools_metrics = memory_metrics.namespace(:pools)
     end
 
     def collect
+      raw = JRMonitor.memory.generate
+      collect_heap_metrics(raw)
+      collect_non_heap_metrics(raw)
+      collect_pools_metrics(raw)
+    end
+
+    private
+
+    ##
+    #  (TODO) usage of increment need to be changed to gauge as soon as this
+    #  type is implemented. This for all the metrics collectors in this class.
+    ##
+
+    def collect_heap_metrics(data)
+      heap = aggregate_information_for(data["heap"].values)
+      heap[:used_percent] = (heap[:used_in_bytes] / heap[:max_in_bytes].to_f)*100.0
```

Comment:
We have to do it, see:

```
irb(main):003:0> (76/100.0)*100
=> 76.0
irb(main):004:0> (76/100)*100
=> 0
irb(main):005:0> (76/100)*100.0
=> 0.0
```

otherwise the division is clearly wrong.